src/encodings/levels.rs

```diff
@@ -107,6 +107,14 @@ impl LevelDecoder {
   }
 
   #[inline]
+  // Set byte array explicitly when start position ('start') and length ('len') are known in advance
```

Comment:
Yes, will do - this coded that way, because of me not understanding those attributes. I will update all similar occurrences.

Could you explain, what `#[inline]` does? Thanks!

Reply:
#[inline] marks the function call as [inline](https://en.wikipedia.org/wiki/Inline_function) so invocation cost can potentially be avoided. This should only be applied to short functions though.